### PR TITLE
Test examples with CI

### PR DESCRIPTION
**Pull request by brikou**
My own project does not work with beta 9 only until beta 6.
TS testing examples will emphasize this, and will allow us to find a fix soon.

### CODE REVIEW

File: examples/githunt-api/package.json [hunk 1]
```diff
@@ -5,9 +5,9 @@
   },
   "dependencies": {
     "apollo-server": "2.2.0-alpha.2",
-    "githunt-api": "apollographql/GitHunt-API.git",
```
Comment (brikou) [inline examples/githunt-api/package.json]: This repository doesn't exist anymore, replaced by a fork.

File: examples/kitchen-sink/package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
     "apollo-server": "^2.2.0",
     "graphql": "^14.0.2",
     "graphql-subscriptions": "^1.0.0",
-    "nexus": "0.12.0-beta.1",
+    "nexus": "0.12.0-beta.6",
```
Comment (brikou) [inline examples/kitchen-sink/package.json]: Unable to upgrade to latest beta without errors

File: examples/ts-ast-reader/src/types/jsdoc.ts [hunk 3]
```diff
@@ -26,7 +26,7 @@ export const JSDocTag = interfaceType({
   },
 });
 
-const jsDocTag = (t: Types.ObjectDefinitionBlock<any>) =>
+const jsDocTag = (t: blocks.ObjectDefinitionBlock<any>) =>
```
Comment (brikou) [inline examples/ts-ast-reader/src/types/jsdoc.ts]: We still have some errors on this example but not sure how to fix others

File: .circleci/config.yml [hunk 4]
```diff
@@ -1,17 +1,87 @@
...
+      - image: circleci/node:10
+    steps:
+      - checkout
+      - run: yarn --frozen-lockfile
+      - run: yarn --cwd=examples/ghost --frozen-lockfile
+      - run: yarn --cwd=examples/ghost tsc
+
+  example_githunt-api:
+    docker:
+      - image: circleci/node:8
```
Comment (brikou) [inline .circleci/config.yml]: This example does not work with node 10

File: examples/star-wars/package.json [hunk 5]
```diff
@@ -7,7 +7,7 @@
   "dependencies": {
     "apollo-server": "^2.2.0",
     "graphql": "^14.0.2",
-    "nexus": "0.12.0-beta.4",
+    "nexus": "0.12.0-beta.9",
```
Comment (brikou) [inline examples/star-wars/package.json]: I wasn't able to use `"nexus": "file:../.."`
We should use `yarn link` on CI, wdyt @Weakky @jasonkuhrt ?

File: .circleci/config.yml [hunk 6]
```diff
@@ -1,17 +1,95 @@
...
+  example_star-wars:
+    docker:
+      - image: circleci/node:10
+    steps:
+      - checkout
+      - run: yarn --frozen-lockfile
+      - run: yarn link
+      - run: yarn --cwd=examples/star-wars --frozen-lockfile
+      - run: yarn --cwd=examples/star-wars link nexus
+      - run: yarn --cwd=examples/star-wars tsc
```
Comment (brikou) [inline .circleci/config.yml]: We should make sure the definition file is generated before running tsc? If you have an idea on how to make this in a clean way @Weakky @jasonkuhrt?

Comment (brikou) [reply .circleci/config.yml]: I'm thinking about running node `dist/schema.js` and then `git diff src/schema.graphql` and  `git diff src/nexus.d.ts` and check that there is no diff.

File: .circleci/config.yml [hunk 7]
```diff
@@ -1,17 +1,99 @@
...
+      - run:
+          command: |
+            rm examples/star-wars/src/star-wars-typegen.ts
+            rm examples/star-wars/star-wars-schema.graphql
+            node examples/star-wars/dist/schema.js
+            git diff --exit-code
```
Comment (brikou) [inline .circleci/config.yml]: here I check that generated files are up to date

File: examples/apollo-fullstack/src/index.ts [hunk 8]
```diff
@@ -15,12 +15,18 @@ const schema = makeSchema({
   types,
   outputs: {
     schema: path.join(__dirname, "../fullstack-schema.graphql"),
-    typegen: path.join(__dirname, "../src/fullstack-typegen.ts"),
+    typegen: path.join(
+      __dirname.replace(/\/dist$/, "/src"),
```
Comment (brikou) [inline examples/apollo-fullstack/src/index.ts]: trick to make sure calling `node dist/schema.js` will find the right dest directory

**Comment by tgriesser** [comment]
Wow, this is awesome stuff @brikou! Thanks for all of the hard work getting this implemented!

Did you figure out what the issue was with your own project?

**Comment by brikou** [comment]
Thanks @tgriesser!
If you change deps to `beta.9` in examples/kitchen-sink/package.json, then you'll see the error I'm facing.

**Comment by jasonkuhrt** [comment]
Great work @brikou 🎉

**Pull request merged**